pkg/securityscan/scanHandler.go

```diff
@@ -211,9 +212,48 @@ func (c *Controller) getDefaultClusterScanProfile(clusterprovider string) (strin
 	if !ok {
 		profileName = cm.Data["default"]
 	}
+	lines := c.splitLines(profileName)
+	if len(lines) > 1 {
+		logrus.Debugf("profilename is splitted by k8s version %v", lines)
+		for _, line := range lines {
+			parts := strings.Split(line, ":")
+			if len(parts) > 1 {
+				k8sRange := parts[0]
+				profile := parts[1]
+				// validate cluster's k8s version matches the profile's k8s version range
+				clusterK8sToMatch, err := semver.Make(clusterK8sVersion[1:])
+				if err != nil {
+					logrus.Errorf("Cluster's k8sVersion is not sem-ver %s %v", c.KubernetesVersion, err)
+					continue
```

Comment:
There is no reason to continue, if the clusterK8sVersion is bad it won't change on the next iteration. Return the error instead of logging it.

Reply:
done